depthai_nodes/ml/messages/img_detections.py

```diff
@@ -226,3 +233,49 @@ def masks(self, value: NDArray[np.int16]):
         masks_msg = SegmentationMask()
         masks_msg.mask = value
         self._masks = masks_msg
+
+    def getVisualizationMessage(self) -> dai.ImgAnnotations:
+        """Returns default visualization message for the detections.
+
+        The message adds the bounding boxes, labels and keypoints to the image
+        annotations.
+        """
+        img_annotations = dai.ImgAnnotations()
+        annotation = dai.ImgAnnotation()
+
+        for detection in self.detections:
+            detection: ImgDetectionExtended = detection
+            rotated_rect = detection.rotated_rect
+            points = rotated_rect.getPoints()
+            pointsAnnotation = dai.PointsAnnotation()
+            pointsAnnotation.type = dai.PointsAnnotationType.LINE_STRIP
+            pointsAnnotation.points = dai.VectorPoint2f(points)
+            pointsAnnotation.outlineColor = OUTLINE_COLOR
+            pointsAnnotation.thickness = 2.0
+            annotation.points.append(pointsAnnotation)
+
+            text = dai.TextAnnotation()
+            text.position = points[0]
+            text.text = f"{detection.label_name} {int(detection.confidence * 100)}%"
+            text.fontSize = 50.5
+            text.textColor = TEXT_COLOR
+            text.backgroundColor = BACKGROUND_COLOR
+            annotation.texts.append(text)
+
+            if len(detection.keypoints) > 0:
+                keypoints = [
+                    dai.Point2f(keypoint.x, keypoint.y)
+                    for keypoint in detection.keypoints
+                ]
+                keypointAnnotation = dai.PointsAnnotation()
+                keypointAnnotation.type = (
+                    dai.PointsAnnotationType.LINE_STRIP
+                )  # change to POINTS when fixed
+                keypointAnnotation.points = dai.VectorPoint2f(keypoints)
+                keypointAnnotation.outlineColor = KEYPOINT_COLOR
+                keypointAnnotation.thickness = 2.0
+                annotation.points.append(keypointAnnotation)
+
```

Comment:
Do we also need to add instance seg mask here?

Reply:
Sadly we cannot do this. The getVisualizationMessage can only return dai.ImgAnnotations or dai.ImgFrame, and not both at the same time. To show also seg masks, we need to make a new HostNode where we return the seg mask.